machine_learning/benchmarks/circuit_qnn_classifier_fit_benchmark.py

```diff
@@ -80,6 +77,55 @@ def parity(x):
             neural_network=self.circuit_qnn, optimizer=self.optimizers[optimizer_name]
         )
 
+    def setup_dataset_iris(self, X, quantum_instance_name, optimizer_name):
+        """Training CircuitQNN function for iris dataset."""
+
+        self.output_shape = 3
+
+        # scaling data
+        scaler = MinMaxScaler((-1, 1))
+        self.X = scaler.fit_transform(X)
+
+        # creating feature map
+        feature_dim = self.X.shape[1]
+        feature_map = ZFeatureMap(feature_dim)
+
+        # creating ansatz
+        ansatz = RealAmplitudes(feature_dim)
+
+        qc = QuantumCircuit(feature_dim)
+        qc.append(feature_map, range(feature_dim))
+        qc.append(ansatz, range(feature_dim))
+
+        def three_class(x):
+            return f"{x:b}".count("1") % 3
+
+        # construct QNN
+        circuit_qnn = CircuitQNN(
+            circuit=qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=three_class,
+            output_shape=self.output_shape,
+            quantum_instance=self.backends[quantum_instance_name],
+        )
+
+        self.circuit_classifier_fitted = NeuralNetworkClassifier(
+            neural_network=circuit_qnn,
+            optimizer=self.optimizers[optimizer_name],
+        )
+
+    def setup(self, dataset, quantum_instance_name, optimizer_name):
+        """setup"""
+        self.X = self.datasets[dataset]["features"]
+        num_inputs = len(self.X[0])
+        self.y01 = self.datasets[dataset]["labels"]
+
+        if dataset == "dataset_synthetic_classification":
+            self.setup_dataset_synthetic(num_inputs, quantum_instance_name, optimizer_name)
+        elif dataset == "dataset_iris":
+            self.setup_dataset_iris(self.X, quantum_instance_name, optimizer_name)
```

Comment:
Same, unify signatures.

Reply:
I have unified the signatures by moving the data scaling (MinMaxScaler) outside of the `setup_dataset_iris` function. Not sure if this is the best way to go about it